pilot/pkg/proxy/envoy/config.go

```diff
@@ -134,23 +134,33 @@ func BuildConfig(config meshconfig.ProxyConfig, pilotSAN []string) *Config {
 
 // buildListeners produces a list of listeners and referenced clusters for all proxies
 func buildListeners(env model.Environment, node model.Node) (Listeners, error) {
+	var listeners Listeners
+
 	switch node.Type {
-	case model.Sidecar, model.Router:
+	case model.Sidecar:
 		instances, err := env.HostInstances(map[string]*model.Node{node.IPAddress: &node})
 		if err != nil {
 			return nil, err
 		}
+
 		services, err := env.Services()
 		if err != nil {
 			return nil, err
 		}
-		listeners, _ := buildSidecarListenersClusters(env.Mesh, instances,
-			services, env.ManagementPorts(node.IPAddress), node, env.IstioConfigStore)
-		return listeners, nil
+
+		listeners, _ = buildSidecarListenersClusters(env.Mesh, instances, services, env.ManagementPorts(node.IPAddress), node, env.IstioConfigStore)
+	case model.Router:
```

Comment:
nit: Lets rename it to gateway please. No one is using router yet.

Reply:
this is actually interesting - we are already squatting a the name `Gateway` in the `model` package

https://github.com/istio/istio/blob/master/pilot/pkg/model/config.go#L323

Router didn't have this problem.

Maybe we just call this Proto `GatewayRules`